app/models/test.rb

```diff
@@ -8,10 +8,18 @@ class Test < ApplicationRecord
   has_many :test_users, dependent: :destroy
   has_many :users, through: :test_users, class_name: 'TestUser'
 
-  def self.sorting_an_array(title_category)
-    joins(:category)
-      .where(categories: { title: title_category })
-      .order(title: :desc)
-      .pluck(:title)
-  end
+  validates :title, presence: true
+  validates :level, numericality: { greater_than: -1 }
+  validates_uniqueness_of :title, scope: [:level]
```

Comment:
[в документации](https://guides.rubyonrails.org/active_record_validations.html#uniqueness) к хелперу `uniqueness` сказано следующее:
> It does not create a uniqueness constraint in the database, so it may happen that two different database connections create two records with the same value for a column that you intend to be unique. To avoid that, you must create a unique index on that column in your database.

другими словами, чтобы эта была настоящая уникальность (не только на уровне приложения, но и на уровне базы), нужно в базе создать уникальный индекс

про правильную миграцию можно посмотреть к примеру на [СО](https://stackoverflow.com/questions/32927262/create-migration-with-uniqueness-to-a-scope)

примерно таким образом:

```ruby
class AddIndexToTests < ActiveRecord::Migration[5.2]
  def change
    add_index :tests, [:title, :level], unique: true
  end
end
```